aml-org/amf-aml
# APIMF-3209: cherry-pick to remod

- Publish version 6.0.0-SUPER-SECRET-SNAPSHOT
- Refactor dialect header matching
- Renamed DialectHelper to AMLParsingHelper
- Reorganized testing classes into new packages
- Refactor Dialect registration logic into new DialectRegistrationHelper in tests
- APIMF-2909:   - Create parsing & rendering plugins for dialects, vocabularies & dialect instances   - Move legacy indexation logic to new configuration   - Disable parsing & rendering from AMLPlugin (canParse=false & canUnparse=false)   - Remove automatic dialect registration & manually registering dialect references in tests   - Partial "forInstance" implementation
- Update dialect registration with new modularization interface
- Extract dialect reference collection from AMLConfiguration to new class
- APIMF-2879: register pipelines in Configuration with key consisting of vendor and pipeline
- avoid loosing previous initialization
- APIMF-2812: update interfaces and adopt core changes
- APIMF-2867: remove generic vendors. Adopt core version for that
- Fix inspections + split classes into separate files
- Hierarchize AML model with HasObjectRange and PropertyLikeMapping
- APIMF-3028: add annotation mapping & extension mapping models for semantic extensions
- APIMF-3028: fix merging test
- APIMF-2990: adjust resolution stage to new interface + renaming (#259)
- APIMF-2978: started registering validation profiles for custom-validation and provided api validations in static environment
- Fixed tests compilation and renamed mispelled variable
- APIMF-2979: validation profiles from dialects are registered in Configuration. Added validation profile creation in forInstance and fixed it to use resolved dialects.
- Added TO-DOs to delete methods allowing decremental building of AMLConfiguration
- APIMF-3048: add javadoc and remove private[amf] (#256)
- remove javadoc links to private classes (they break sbt compile doc)
- Refactor: extract property term & range parsing to separate classes
- Add node mappable model to model hierarchy
- Renamed extension mapping to semantic extension
- APIMF-3035: semantic extension parsing basic implementation
- APIMF-3086: add links flag to sbt and fix javadocs
- refactor: adjust transformation step interface to make client conversion possible
- APIMF-2995: exported interfaces for AMLConfiguration and AMLClient
- refactor: renaming package with client interfaces to amf.client.exported
- Fix annotation for PropertyMapping on dialect parser
- Fix tests
- APIMF-2964: dialect namespace generation plugin implementation, stored when registering dialect
- refactor: rename staticAliases to defaultAliases
- Rename annotation mappings field taget -> domain
- Lift PropertyMapping fields to PropertyLikeMapping level
- Parsing extra facets from annotation mappings + small parsers refactor
- APIMF-2955: added support for allowMultiple in dialect instances
- APIMF-2896: add scaladoc to remod/exported files
- Added annotations for 'external' items
- Added annotations for 'external' entry
- Added annotations for literal ranges
- Added lexical for resolved nodeMapping range references
- Added annotations to property term parsing
- Fixed semantic extension object class, it returns StrField instead of AnnotationMapping. Added id resolution for annotation mapping ranges
- APIMF-3037: added annotation mapping and semantic extensions emission. Tests fail due to lines shifting because there are no annotations
- Test: Restricted test to jvm due to minor changed in how numbers get emitted in JVM vs JS
- Added SLF4J as test dependency to omit annoying logs because of shacl
- Fix: implemented classConstructor method for AnnotationMapping
- FromUnionNodeMapping annotation
- Adopt fixes from core about amf plugins and error handling
- APIMF-2878: extracted entities and serializable annotations and added to new registry
- Fix rebase errors
- FIx compile for test and improve some dialect registration handling
- fixes: minor adjustments due to change in ParseConfiguration apply
- APIMF-2878: added entities and serializable search. Added facades in ParserConfig and the config inside the ParserContext
- APIMF-2878: fix test compilation
- Adopt core changes. Add improve rendering interface. Use configuration for rendering and parsing. Stop consuming static config
- define withCustomValidationEnabled method
- APIMF-2984: implemented withCustomProfile in AMLConfiguration and added AMLLegacyValidationPlugin to AML predefined plugins
- Tests: fixed several AML tests due to validation or usage of legacy AMLPlugin register
- Test: erased legacy initialization as it is no longer necessary
- avoid registering static validations through platform
- Adopt core changes to fix tests
- Add methods for parse and index validation profiles
- Tests + Refactor: changed tests to use AMLConfiguration. Also removed DialectRegistry usages and replaced it with the NodeMappableFinder to propagate known dialects.
- Tests + Refactor: removed AMLPlugin and fixed tests.
- Cleanup: removed DialectsRegistry
- Extracted ShaclValidation
- Tests: Fixed custom validation tests
- rebase fixes: adjust legacy error handlers in tests and compile issues
- APIMF-2984: refactored validation plugins to clean out legacy stuff
- APIMF-3132: replace ValidationReport for list of results in AMFResult (#287)
- APIMF-3098: add support for partial parsing of the flatten graph
- APIMF-3036: implemented extension finder, registered semantic extensions in EntitiesRegistry
- APIMF-2982: change carried from amf-core
- Uncomment withErrorHandlerProvider() method
- Adopt env removal from core
- APIMF-2994: delete legacy parsing/rendering options
- APIMF-2997: added corrections for JS bundle
- Refactor dialect reference resolution
- Add annotation mappings to reference resolution stage
- Adopt new modules from core
- Annotations for fields now include the whole entry, not only value
- APIMF-3133: adjust pipelines names to use mediaType
- refactor: adjust usages of PipelineId object
- APIMF-3159: refactor packages in aml (#302)
- Adopt APIMF-3156
- move scala resource loader to client.scala.resource
- Beta fixes: removed namespaces from top level exports
- APIMF-3091: added explicit option to emit json-ld in embedded form in tests
- APIMF-3008: propagated EventListeners and replaced ExecutionLog with new events
- APIMF-3131: add method to create empty AML config
- APIMF-3157: add setter of ExecutionEnv in configuration
- fixes: remove global execution context import in AMLConfiguration
- APIMF-3206: removed withCustomValidations support in AMF
- Manual revert of APIMF-2980 due to performance problems. This was found in release 4.7.4 and was just reverted now. It isn't automated as file content and location has changed
- APIMF-3171: exclude Tomcat depedency
- APIMF-3208: create AMLElementClient and redefine AMLBaseUnitClient (#325)
- APIMF-3168: added client interfaces to search dialects and semantic extensions
- APIMF-3203: expose all safe usages of global context, adjust renaming
- APIMF-3202/0: Deleted AMF Logger
- APIMF-3202/1: Renamed bu for baseUnit
- APIMF-3202/2: added method to AMLConfig to be coherent with AMFGraphConfig
- Fixed rebase changes
- APIMF-3024: implements forInstance function for nested dialect references
- APIMF-3174, APIMF-2996: check client interfaces with internal functionality, remove scala imports from platform
- APIMF-2996: add missing exports and functions in platform interfaces
- adjust imports for pipeline/stages package refactor in core
- refactor: improve CustomShaclValidator
